echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/config/ArtifactTemplateProperties.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2018 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.echo.config;
+
+import com.netflix.spinnaker.echo.pipelinetriggers.artifacts.CustomJinjaTemplate;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.hibernate.validator.constraints.NotEmpty;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.validation.annotation.Validated;
+
+import javax.validation.Valid;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Data
+@Configuration
+@ConfigurationProperties(prefix = "artifacts")
+@Validated
+public class ArtifactTemplateProperties {
+  @Valid
```

Comment:
I'm not familiar with this annotation, what is its purpose?

Reply:
Per the [Spring docs](https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-external-config.html#boot-features-external-config-validation) "Although nested properties will also be validated when bound, it’s good practice to also annotate the associated field as @Valid. This ensure that validation is triggered even if no nested properties are found."

So my understanding is this ensures we validate the field in all cases...though I'll be honest I'm not sure I 100% understand the subtletly there and am just following the best practice.  (If we don't bind anything to the property, I'm not sure why we'd need to validate...)